### PR TITLE
test: added irrelevant entities tests

### PR DESCRIPTION
**Pull request by DoctorVoid**

### CODE REVIEW

File: test/integration-tests/test-server/client.ts [hunk 1]
```diff
@@ -7,6 +7,10 @@ export const url = `http://localhost:${polarisPropertiesPath.port}${
 
 export const graphqlRequest = async (data: string, headers: any, variables: any) => {
     const graphQLClient = new GraphQLClient(url, { headers });
-    const result = await graphQLClient.request(data, variables);
-    return result;
+    return graphQLClient.request(data, variables);
+};
+
+export const graphqlRawRequest = async (data: string, headers: any, variables: any) => {
```
Comment (osher-sade) [inline test/integration-tests/test-server/client.ts]: make `variables` param to be undefined by default

File: test/integration-tests/tests/irrelevant-entities-tests.test.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import { graphqlRawRequest } from '../test-server/client';
```
Comment (yarinvak) [inline test/integration-tests/tests/irrelevant-entities-tests.test.ts]: irrelevant-entities-tests file name change
make standard in all tests files

File: test/integration-tests/test-server/db-preparation.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@ const realityIdHeader = (realityId: number) => ({ headers: { realityId } });
 export const prepareDb = async () => {
     const books = [];
     for (let i = 0; i < titles.length; i++) {
-        books.push({ author: 'chen', title: titles[i], id: i });
+        books.push({ author: 'arik', title: titles[i], testId: i, dataVersion: i + 1 });
```
Comment (yarinvak) [inline test/integration-tests/test-server/db-preparation.ts]: split db preparation files to each int test and init db with the int tests own entities

File: test/integration-tests/tests/irrelevant-entities-tests.test.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    test('2 irrelevant entities not starting with f', async () => {
+        const queryBook = `{
+          booksStartsWith(startsWith:"f"){
+            id,
+            title
+          }
+        }`;
+        const { extensions }: any = await graphqlRawRequest(queryBook, headers, {});
+
+        expect(extensions.irrelevantEntities.booksStartsWith.length).toBe(2);
```
Comment (yarinvak) [inline test/integration-tests/tests/irrelevant-entities-tests.test.ts]: also check the entities that are returned equal to the expected entities

Comment (yarinvak) [reply test/integration-tests/tests/irrelevant-entities-tests.test.ts]: in all tests

File: test/integration-tests/tests/irrelevant-entities-tests.test.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        const queryBook = `{
+          booksStartsWith(startsWith:"f"){
+            id,
+            title
+          }
+        }`;
+        const { extensions }: any = await graphqlRawRequest(queryBook, { 'reality-id': 1 }, {});
+
+        expect(extensions).not.toBeDefined();
+    });
```
Comment (yarinvak) [inline test/integration-tests/tests/irrelevant-entities-tests.test.ts]: check that no mgf test is missing here

File: test/integration-tests/test-server/db-preparation.ts (hunk 3 above)
Comment (yarinvak) [reply test/integration-tests/test-server/db-preparation.ts]: its copying the same code over and over, maybe change the name to basic db preparations? for more special case tests then u will do also a separate initialization?

> split db preparation files to each int test and init db with the int tests own entities

its copying the same code over and over, maybe change the name to basic db preparations? for more special case tests then u will do also a separate initialization?

Comment (DoctorVoid) [reply test/integration-tests/test-server/db-preparation.ts]: > split db preparation files to each int test and init db with the int tests own entities

its copying the same code over and over, maybe change the name to basic db preparations? for more special case tests then u will do also a separate initialization?

File: test/integration-tests/tests/irrelevant-entities-tests.test.ts (hunk 4 above)
Comment (DoctorVoid) [reply test/integration-tests/tests/irrelevant-entities-tests.test.ts]: but they are generated in runtime (guid) i cant expect them

Comment (yarinvak) [reply test/integration-tests/tests/irrelevant-entities-tests.test.ts]: you can check other properties

File: test/integration-tests/test-server/db-preparation.ts (hunk 3 above)
Comment (yarinvak) [reply test/integration-tests/test-server/db-preparation.ts]: ok

**Pull request merged**